baseframe/forms/validators.py

```diff
@@ -238,8 +238,9 @@ def check_url(self, invalid_urls, url, text=None):
                     # For text patterns, do a substring search. For regex patterns (assumed so if not text),
                     # do a regex search. Test with the final URL from the response, after redirects,
                     # but report errors using the URL the user provided
-                    if (isinstance(pattern, basestring) and pattern in rurl) or (pattern.search(rurl) is not None):
-                        return message.format(url=url, text=text)
+                    if (isinstance(pattern, basestring) and pattern in rurl) or (
+                        isinstance(pattern, re._pattern_type) and pattern.search(rurl) is not None):
```

Comment:
Try `if pattern in rurl if isinstance(pattern, basestring) else pattern.search(rurl) is not None:`